<!DOCTYPE html>
<html lang="en">
	<head>
		<title>Peptide De novo sequencing</title>
		<meta http-equiv="content-type" content="text/html; charset=utf-8" />
        <meta charset="UTF-8">
        <meta name="description" content="Discover the fascinating world of peptide de novo sequencing. Learn about the essential ingredients of life and how proteins, the microscopic marvels, contribute to our existence. From milkshakes to your hair and skin, proteins are the versatile building blocks that sustain life. Uncover the genetic code within our DNA, akin to a vast library of cookbooks, housing the recipes for over 10,000 vital proteins. Join Aditi Sharrma and Prajwal DSouza in unraveling the intricate processes that keep our bodies functioning harmoniously every single day.">
        <meta name="keywords" content="plots.js, prajwalsouza, Prajwal DSouza, template">
        <meta name="author" content="Aditi Sharma, Prajwal DSouza">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">

		<link href="https://fonts.googleapis.com/css?family=Nunito&display=swap" rel="stylesheet">
		<link href="https://fonts.googleapis.com/css?family=Raleway:100,200,300,400,500,600,700,800,900&display=swap" rel="stylesheet">
        <link href="https://fonts.googleapis.com/css2?family=Gaegu:wght@300;400;700&display=swap" rel="stylesheet">

		<!-- Some times used library for icons -->
		<link href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous">
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
		<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>


        <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/prajwalsouza/IgnusTheme@1.3/ignus-theme.css">
        <script src="https://cdn.jsdelivr.net/gh/prajwalsouza/IgnusTheme@1.3/ignus-theme.js"></script>


		<link rel="icon" href="square.png">
		
		<script type="application/ld+json">
			{
			  "@type": "blog",
			  "name": "Peptide De novo sequencing",
			  "author": "Aditi Sharma, Prajwal DSouza",
			  "description": "Discover the fascinating world of peptide de novo sequencing. Learn about the essential ingredients of life and how proteins, the microscopic marvels, contribute to our existence. From milkshakes to your hair and skin, proteins are the versatile building blocks that sustain life. Uncover the genetic code within our DNA, akin to a vast library of cookbooks, housing the recipes for over 10,000 vital proteins. Join Aditi Sharrma and Prajwal DSouza in unraveling the intricate processes that keep our bodies functioning harmoniously every single day."
			  }
		</script>

        <link rel="stylesheet" href="/css/universal-parabolic-constant/universal-parabolic-constant.css"/>

        <!-- Google tag (gtag.js) -->
        <script async src="https://www.googletagmanager.com/gtag/js?id=G-MNFFCG66EJ"></script>
        <script>
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());

            gtag('config', 'G-MNFFCG66EJ');
        </script>
      

	</head>
	<body style='margin:0px; display: block;'>
		<div id="containAll" class="p-sm-3 p-1" style=" overflow: hidden; ">
            <div style="border: 1px solid hsla(0,0%,0%, 0); border-radius: 10px; background: var(--generalBackground); ">
                
                <header id="upc-header" class="pr-2 pr-sm-3" style="position: fixed;font-family: Raleway; overflow: hidden; width: calc(100%); top: 0; z-index: 3; opacity: 0;" onclick="denovoApp.headerClick()">
                    <div style="background-image: linear-gradient(to bottom, var(--generalBackground),var(--generalBackground), transparent);color: hsl(0,0%,20%);font-size: small;min-height: 131px;" class="text-ignus p-4 pl-5 pr-5 p-md-3 pl-md-4 pr-md-4 text-md-left text-center">
                        <span style="cursor: pointer;"> 
                            <b id="currentMenuName" style="color:hsla(198, 0%, 37%, 1)">What's the problem?</b><br class="d-inline-block d-md-none"/>
                            <span style="font-size:x-small">The Parabolic Constant Mystery</span>
                        </span>
                        <br/>
                        <span id="upc-menus" style="display: none;"></span>
                    </div>
                    
                </header>

                <main>
                    <div class="w-100 text-center text-ignus pt-5 mt-3 pb-4" style="font-family: Raleway;">
                        <div class="pl-3 pr-3">
                            <b style="font-size: x-large;">Peptide De Novo Sequencing </b><br/>
                            <b style="font-size: large; color: gray">What are the ingredients of that delicious pizza?</b><br/>
                            <p style="font-size: medium; color: hsl(0,0%,30%)">Aditi Sharrma, Prajwal DSouza</p>
                        </div>
                    </div>
    
                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>
                                Proteins, the mighty microscopic marvels present in what we eat, hence, what we are. In milkshakes ü•§, ice creams üç¶... your hair, your skin‚Ä¶ you üë§. These molecular machines are present in all living things, from viruses to the dog üê∂ that barked at you the other day. They are so essential that they keep the grand show of life going. Proteins are like the ultimate multi-taskers of the cellular world. They're the tiny construction workers üèóÔ∏è, dutiful soldiers, speedy messengers, vigilant guards and the skilled repair crew üîß that keep our bodies running smoothly day in, day out.<br/><br/>

                                They're crafted by our cells using the blueprints written in our DNA. Heard of the genetic code in the DNA? This code is like a huge library üìö, stuffed with cookbooks. These cookbooks contain genes, which are like recipes for whipping up every protein your body needs. And there are at least 10000 different proteins keeping you alive.
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/dna-to-proteins.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/dna-to-proteins-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-explaining-block">
                            <p>
                               
                                These proteins are built inside your cells using tiny molecules called amino acids üß™, which are like the ingredients in your recipe. These cells are guided by the instructions in the DNA on how to put together particular proteins. Picture a cake üç∞, it needs flour, eggs, sugar, and butter, all in specific quantities and added in a certain order. Similarly, proteins need specific amino acids in the right order to be cooked up right. DNA has the recipe, cookbooks üìö. Amino acids are the ingredients. Proteins are the dishes üçΩÔ∏è. Hungry yet? üòã 

                                <br/><br/>
                                As an example, the following genetic code in your DNA, could be the cookbook instruction üìñ for creating the dish (protein) we want.
                                <br/><br/>
                                <span class="text-ignus text-large-ignus"> 
                                    TGT - <span class="p-lightBlue">TAC</span> - ATT - CAA - AAT - TGT - CCT - CTC - GGT
                                </span>
                                
                                <br/><br/>
                                These instructions would lead the cells to place amino acids (ingredients) one after another leading to a molecular chain of amino acids placed in the following order.
                                <br/><br/>
                                <span class="text-ignus" style="font-size: large;">
                                    Cysteine - <span class="p-lightBlue">Tyrosine</span> - Isoleucine - Glutamine - Asparagine - Cysteine - Proline - Leucine - Glycine
                                </span><br/>
                                <span class="text-ignus" style="font-size: x-large;">
                                    C<span class="p-lightBlue">Y</span>IQNCPLG
                                </span>
                                
                                <br/><br/>
                                In this case, the instruction earlier in the DNA was for a small protein (technically peptide, but we will get to that.) called <span class="p-lightPink">Oxytocin</span> üòç, often referred to as the "love hormone" or the "cuddle chemical." It plays a crucial role in social bonding, fostering feelings of trust, empathy, and connection. The amino acid sequence for this <span class="p-lightPink">oxytocin</span> is <span class="p-lightPink">CYIQNCPLG</span>.<br/>
                                There are over 20 amino acids, and only using 20 of these, our bodies create 10,000 different types of proteins, with each protein playing a different role. For example, hemoglobin protein üíâ is made by the cells in your bone marrow, because the cells are instructed by your DNA to do so, which is then released into your bloodstream. Tyrosinase is another protein, which is responsible for producing melanin, which gives your hair the color üåà. Loss of this protein, or cells choosing not to produce it can lead to loss of hair color.
                                <br/><br/>
                                
                                While <span class="p-lightPink">oxytocin</span> is just 9 amino acids, hemoglobin is about 300 and Tyrosinase is over 500 amino acids long. 
                                
                            </p>
                        </div>
                    </div>

                    <div class="w-100 pl-4 pt-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center pb-3">
                        <div class="our-text-block text-center pt-4">
                            <p class="p-White" style="font-size: large; font-family: Raleway;">
                                <b>So, What is de novo sequencing?</b>
                            </p>
                        </div>
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                Here's the big challenge. Let's call it "de novo sequencing" which is a fancy way of saying "figuring out the ingredients of a protein by tasting it" üïµÔ∏è‚Äç‚ôÄÔ∏èüîç. It's like heading to a restaurant, eating a lip-smacking dish üç≤, and then trying to identify the ingredients just based on your taste! Sounds fun but tough, doesn't it? The goal is to identify the composition and sequence of the protein. That is, to identify the amino acid pattern.  <br/><br/>
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/what-is-denovo.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/what-is-denovo-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                At this point, allow me to introduce the superstars of our story ‚Äî peptides. Peptides are like mini proteins. Think about a giant pizza üçï(that's your protein), peptides are like those cute mini pizza bites, smaller in size but oh-so-crucial. Peptides are just shorter chains of these all-important amino acids. Remember oxytocin, that's a peptide, a mini protein. üíñ
                                <br/><br/>
                                From now on, we're shining the spotlight üéØ on these little strands of amino acids, the peptides. Let's think of them as the underdogs, small but mighty, and loaded with info about our bodies. üèãÔ∏è‚Äç‚ôÄÔ∏è 
                                <br/><br/>

                                So, why the heck do we want to figure out the sequence of the peptide? ü§∑‚Äç‚ôÄÔ∏è Well, if we crack the sequence of amino acids - the 'secret ingredients' üß™ - in our peptide, we unlock a whole treasure chest üóùÔ∏èüîì of information about what it is and what it does. It's like decoding a secret message written in an alien language! üëΩüìú
                                <br/><br/>

                                By identifying a peptide in a sample, we could potentially develop a new wonder drug üíä that uses the peptide, or a treatment that blocks it. It's like having a secret weapon in the battle against diseases! üõ°Ô∏è We could even learn something groundbreaking about how the body works! üß†‚ö° It's like discovering a new law of physics... but inside our bodies! How cool is that? ü§ìüöÄ
                            
                            </p>
                        </div>
                        
                    </div>
                    <div class="w-100 pl-4 pt-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center pb-3">
                        <div class="our-text-block text-center pt-4">
                            <p class="p-White" style="font-size: large; font-family: Raleway;">
                                <b>So, how do we do it? </b>
                            </p>
                        </div>
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                Well, we could just squint really hard at them under a microscope ü¶†üî¨. But trust me, even if you have the eyes of an eagle, you wouldn't see much. They're way too small. Sure, there have been some incredible advances recently <a href="https://phys.org/news/2021-11-dna-proteins-de-novo-designed-nanopore.html" target="_blank">with techniques</a> that read each amino acid, but it's still not always possible.<br/><br/>
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                How about just weighing it? üèãÔ∏è‚Äç‚ôÄÔ∏è If my peptide weighs, let's say, 200 (in super-simplified units), then I could whip out a catalog üìö and see that 200 is the mass of oxytocin. So, it must be oxytocin, right? Ehhh... not so fast. There could be a whole bunch of other peptides out there that also weigh 200. Worse, what if this peptide is from a snake venom üêç that's never been observed before and hence, doesn't even make it to the catalog?<br/><br/>

                                Aha! Scientists early on had a light bulb üí° moment. They thought, "Let's break the peptide into pieces, then weigh the pieces." It's like disassembling a Lego tower to understand how it was built.<br/><br/>

                                But then, a new challenge raises its ugly head üê≤. How do we weigh so many tiny pieces at once? It's like trying to weigh a bunch of confetti üéâ - all at the same time!<br/><br/>

                                

                                
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/specta-weighing.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/specta-weighing-vertical.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                Enter the unsung heroes of our story - mass spectrometers üå†. These are like high-tech super scales üß±‚öñÔ∏è that can weigh a ton of things all at once.
                                <br/><br/>

                                
                            
                            </p>
                        </div>
                        
                    </div>
                    

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/mass-spectrometer-mixture-weighing.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/mass-spectrometer-mixture-weighing-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                But wait, there's more! We usually measure a lot of things at once, and mass spectrometers don't just stop at weighing. They also measure quantities üî¢. It's like a super-smart scale saying... "Oh, your sample had more M&Ms üç´ than Coffee beans ‚òï in your Mocha coffee bean cookie üç™." How cool is that!?
                                <br/><br/>

                                
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                Let's say we've got our hands on a peptide called CAT (not the furry creature üê± but a peptide whose amino acid sequence is C, A, and then T). In fact, let's say we have a whole pile of them, like a giant CAT party üéâ. But before we can start the weighing party, we've got to break them into smaller parts. Kind of like smashing a pi√±ata üéä. And we have some pretty fancy ways to do this (imagine sophisticated scientific pinata smashers like CID, ETD etc.).
                                <br/><br/>
                                When we bash our CAT peptide, it can break into C, CA pieces.

                                <br/><br/>

                                
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/breaking.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/breaking-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                ADITI can break into A, AD, ADI and ADIT and so on. 

                                <br/><br/>

                                
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                So, let's say, we've got our CAT peptide broken into <span class="p-White">C</span> and <span class="p-White">CA</span> parts. Now, when we feed these pieces into our trusty mass spectrometer to weigh, we should see two distinct signals üìàüìâ. One corresponds to the smaller <span class="p-White">C</span> piece and another to the larger <span class="p-White">CA</span> piece.
                                <br/><br/>

                                But hold on, what about the whole, unbroken CAT peptide? Well, it can still hang around in our sample, giving us a third signal üìä.
                                <br/><br/>

                                So, in the end, we've got three peaks on our mass spectrometer's readout, one for each of the <span class="p-White">C</span>, <span class="p-White">CA</span>, and the intact <span class="p-White">CAT</span>.


                                <br/><br/>

                                
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/cat-basic.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/cat-basic-vertical.png" width="100%">
                        </div>
                        
                    </div>


                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                For example, talking about CAT, we know that amino acid C weighs about 103 giving us a peak at 103. Similarly, we know that A weighs about 70, so the mass of CA must be 103 + 70 = 173, where we see the second peak. The mass of amino acid T is about 100, giving us another peak from CAT, at 103 + 70 + 100 = 273. 
                                <br/><br/>

                                The same process happens with a peptide called ADITI. The peak at 300 is due to ADI, and as T is about 100, we get a peak at 400 corresponding to ADIT. 

                                <br/><br/>

                                
                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/aditi-basic.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/aditi-basic-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                The values can be generated for any sequence. Here's a tool to try it out yourself. (use b ions checkbox only in the tool, this will be explained in detail later) <br/> <br/>

                                <a href="http://db.systemsbiology.net:8080/proteomicsToolkit/FragIonServlet.html" target="_blank">Fragment Ion Calculator - systemsbiology.net</a>
            
                                <br/><br/>

                                But let's go back to CAT..  Let's say, you didn't know this was CAT peptide and we told you that this peptide had three amino acids. Can you figure out the amino acid sequence? Are you ready to perform De Novo Sequencing?

                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pt-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center pb-3">
                        <div class="our-text-block text-center pt-4">
                            <p class="p-White" style="font-size: large; font-family: Raleway;">
                                <b>Let's do De Novo Sequencing </b>
                            </p>
                        </div>
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                So, you have taken an unknown peptide, smashed it and then sent it into the mass spectrometer. The mass spectrometer has spit out a spectra with three humps. Time to figure out what was in the sample. <br/>To begin, let's just call the unknown peptide X<sub>1</sub> X<sub>2</sub> X<sub>3</sub>. 

                            
                            </p>
                        </div>
                        
                    </div>

                    


                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/cat-denovo-basic.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/cat-denovo-basic-vertical.png" width="100%">
                        </div>
                        
                    </div>


                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                The furthest peak must be from the largest mass, X<sub>1</sub> X<sub>2</sub> X<sub>3</sub> itself, the unbroken peptide.  <br/><br/>
                                
                                So, X<sub>1</sub> X<sub>2</sub> X<sub>3</sub> weighs 273. This cannot directly tell us anything, as we know the masses of amino acids only. And yes, we are approximating here for simplicity. 


                            
                            </p>
                        </div>
                        
                    </div>

                    
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/animo-acid-list.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/animo-acid-list-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                One can see that the peak for the smallest mass is 103 which should be the smallest fragment. So in our case, X<sub>1</sub> is the smallest fragment and must have a mass of 103. This is a single amino acid. But which is it? This is when you go through the amino acid list. C is an amino acid with a mass of 103, so <span class="p-lightBlue">X<sub>1</sub></span> must be <span class="p-lightBlue">C</span>. <br/><br/>

                                The next peak is at 173 and it must be from <span class="p-lightBlue">X<sub>1</sub></span> X<sub>2</sub>. So, <span class="p-lightBlue">X<sub>1</sub></span> X<sub>2</sub> weighs 173. <br/><br/>

                                This tells us that X<sub>2</sub> must be an amino acid with a weight 70. Go through the amino acid list, there is an amino acid corresponding to 70, it's A, Alanine. So, <span class="p-lightOrange">X<sub>2</sub></span> must be <span class="p-lightOrange">A</span>. And finally, the peak at 273 must be from the whole peptide <span class="p-lightBlue">X<sub>1</sub></span> <span class="p-lightOrange">X<sub>2</sub></span> X<sub>3</sub>, so since <span class="p-lightBlue">X<sub>1</sub></span> <span class="p-lightOrange">X<sub>2</sub></span>  is 173, X<sub>3</sub> must be an amino acid with mass 100.. Which our list tells us, <span class="p-lightPink">X<sub>3</sub></span> must be <span class="p-lightPink">T</span>.  

                            
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/cat-denovo-calc-basic.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/cat-denovo-calc-basic-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                So.. we now know <span class="p-lightBlue">X<sub>1</sub></span> <span class="p-lightOrange">X<sub>2</sub></span> <span class="p-lightPink">X<sub>3</sub></span> is <span class="p-lightBlue">C</span><span class="p-lightOrange">A</span><span class="p-lightPink">T</span>. Congrats. That's your first de novo sequencing. 

                                <br/><br/>
                                Now that you know, wanna try out another?
                                Here's some help, this one is 5 amino acids long peptides. 

                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/team-denovo.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/team-denovo-vertical.png" width="100%">
                        </div>
                        
                    </div>



                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                And the solution is‚Ä¶ 

                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/team-denovo-calc.png" width="100%">
                        </div>
                        
                    </div>
                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex d-md-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/team-denovo-calc-vertical.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                TEAM. <br/><br/>
                                CEAM? That's possible too.. If we consider 102 to be associated with C. This is a very simplified version. Because, isn't T, 101.04?  <br/>
                                Yep, you caught us! üé£ We did some rounding up, sort of like rounding up œÄ to 3 (although not quite as dramatic!). However, it's important to note there's another simplification here that matters a lot. The ends of these peptides - imagine them like the caps on a tube of toothpaste ü¶∑ - can have some effects on the values we see in the spectra. <br/> <br/>
                                But hey, before we dive deeper into that, let's take a breather and try some fun puzzles üß©. For now, let's simplify things and replace the peak curves with simple bars (kind of like a bar graph üìä), and maybe we'll skip over the intensities too. Your mission, should you choose to accept it, is to place the appropriate amino acids that match the bars. Ready? Set. Go! üèÅ
                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" style="height: 700px; width: 100%; position: relative;">
                            <div id="animo-acid-dragdrop-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                        </div>
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                Let's play.
                                Here's some more examples to try out. 

                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" style="height: 700px; width: 100%; position: relative;">
                            <div id="main-game-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                        </div>
                    </div>

                    
                    <div class="w-100 pl-4 pt-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center pb-3">
                        <div class="our-text-block text-center pt-4">
                            <p class="p-White" style="font-size: large; font-family: Raleway;">
                                <b>Simple isn't it? But Actually.. </b>
                            </p>
                        </div>
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                So.. things cannot be this simple right? Yep. There's more. 
                                What does it mean for the amino acid C (Cysteine) to have a mass of 103? Turns out, we are actually dealing with something called Residue mass. It will make sense in a moment. 
                                The amino acids have a particular structure. Let's look at Cysteine. There is a side chain called RC and for Cystine, it is CH2 - SH. This varies depending on the amino acid. There is an N-terminal (NH2) and a C terminal (COOH). This is when our amino acid is alone all by itself. 

                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/just-amino-acid.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                But, let's just say that it is joined by two new amino acids, A and E with side chains RA and RE forming a small peptide. Then, there is one H (Hydrogen) lost at the N terminal and an OH lost at the C terminal. Basically, a water molecule is lost. The residue mass of C is this mass, the whole amino acid mass, minus the water molecule that is lost. Similarly, we can see the residue of A and E. The residue contains the N and C end, but minus the H and OH. 

                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/amino-acids-joined.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                So, when fragmented by collision or other method before weighing them in the mass spectrometer. This collision often leads to a charge on the fragment of +1, adding a hydrogen to the N terminal. So.. Consider the peptide FAC above, If you want the mass of fragment FAC, you need to take the residual mass of F, A and C.. but also the extra hydrogen added to the N terminal of F. 

                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/fragment-mass-calculation.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                So, the mass of the fragment is ‚Äúshifted‚Äù by one unit.   

                            </p>
                        </div>
                        
                    </div>

                    <div class="w-100 pt-5 pb-5 pl-4 pr-4 pl-sm-5 pr-sm-5 d-md-flex d-none justify-content-center">
                        <div class="our-text-block text-center">
                            <img src="/assets/images/denovo-blog/fragment-mass-calculation-blocks.png" width="100%">
                        </div>
                        
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>

                                Hope, why the spectra earlier was showing a peak at 102 instead of 101 for T, now makes sense. (and 231 for TE and 302 for TEA in the earlier section). 

                                <br/><br/>

                                For more details on structures of other amino acids, you can look into 
                                <a href='https://en.wikipedia.org/wiki/Amino_acid#/media/File:ProteinogenicAminoAcids.svg' target="_blank">wikipedia</a>.

                            </p>
                        </div>
                        
                    </div>

                    
                    





                    <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" style="height: 400px; width: 100%; position: relative;">
                            <div id="upc-measurement-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                        </div>
                    </div>
    
                    
    
                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center">
                            <p>This might seem like a co-incidence. But, we look for a circle when <span class="p-White" > œÄ</span>  pops up in various places right? Just check <a class="p-lightBlue" href="https://www.youtube.com/watch?v=HEfHFsfGXjs" target="_blank" >3B1B</a> :P  So, why not dig a little deeper. And if we explore a little more, there are other seemingly unrelated math problems where you see this number show up. </p>
                        </div>
                    </div>
    
                    
    
                    <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" style="height: 400px; width: 100%; position: relative;">
                            <div id="distance-between-any-two-points-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                        </div>
                    </div>
    
                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center">
                            <p>
                                
                                So, What do you think is the average <span class="p-lightPink" >distance </span>between any two points inside a <span class="p-lightYellow" >unit square</span>? Take a guess. :D
                            </p>
                            <div class="mt-5">
                                <input type="range" min="0" max="2" value="0.8" step="0.001"  class="commonSlider mb-3" id="distance-between-any-two-points-Slider">
                                <div class="d-block d-sm-none"></div>
                                <label id="distance-between-any-two-points-SliderLabel" class="p-lightPink pl-sm-3 pl-0" for="distance-between-any-two-points-Slider" style="min-width: 100px;" >0.8</label>
                                <div>
                                    <p id="distance-between-any-two-points-SliderInfo" class="text-small-ignus" style="color: hsl(0deg 0% 100%);">
                                        Move the <span class='p-lightPink'>slider</span> to choose your answer...  and also, the points <span class="p-lightBlue">A</span> and <span class="p-lightPurple">B</span> above are draggable. :)
                                    </p>
                                    
                                    <div id="gph2-checkAnswerButton" class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus" style="border: 1px solid hsla(var(--lightPinkH), var(--lightPinkS), var(--lightPinkL), 0.2); color: hsla(var(--lightPinkH), var(--lightPinkS), var(--lightPinkL), 1);">Check</div>
                                    <div id="reveal-1-continueButton" class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus">Continue</div>
    
                                </div>
                            </div>
                            
    
                        </div>
                    </div>
    
                    <div  class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center hidden-element reveal-1 revealable">
                        <div class="our-text-block text-center">
                            <p class="mt-5">
                                It is about  <span class="p-lightPink" >0.52140.</span> To be more precise, it is. <br/><br/>
                                
                                <span class="p-lightPink" >
                                    \(\color{#e3a0cd}{8\left(0.0416{\color{#ffffff}{P}}-0.030473\right)}\)
                                </span><br/>

                                And exactly, it is

                                <div style="font-size: 25px;">
                                    \(\color{#e3a0cd}{8\left(\frac{\color{#ffffff}{P}}{24}-\frac{\sqrt{2}}{30}+\frac{1}{60}\right)}\)
                                </div>
                                
                                
                                <br/>
                                <p>
                                Yep. Strange combination of numbers. There's a <span class="p-White" >P</span> there. <br/>

                                There's many more instances where we see this. But, mostly involving a unit square.</p>
                            </p>
                        </div>
                    </div>
                    
                    <div  class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center hidden-element reveal-1 revealable">
                        <div class="our-text-block text-center">
                            <p>
                                <span class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus" onclick="denovoApp.chooseSection(1, 1)" id="rotatingSectionButton-1-1"> Center-Point Distance</span>
                                <span class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus inActiveSectionButton" onclick="denovoApp.chooseSection(1, 2)" id="rotatingSectionButton-1-2"> Corner-Perimeter Distance</span>
                            </p>
                        </div>
                    </div>

                    <div  class="hidden-element reveal-1 revealable" id="extender-section-1" style="max-height: 100px; overflow: hidden; position: relative;">
                        <div style="position: absolute; background: linear-gradient(to bottom, transparent, var(--generalBackground)); pointer-events: none; z-index: 1;" class="w-100 h-100" id="extender-section-1-cover"></div>
                        <div style="position: absolute; bottom: 0px; height: 30px; " class="w-100 text-center dragDownExtension" onclick="denovoApp.extendSection(1)">
                            <span><i id="extender-section-1-chevron-down" class="fa fa-angle-double-down" style="font-size: 23px;"></i></span>
                        </div>
                        
                        <div id="rotatingSection-1-1" class="rotatingSection-1">
                            <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                                <div class="our-text-block text-center pb-1">
                                    <p>
                                        Here is a slightly similar problem. <br/>
                                        What do you think is the average <span class="p-lightPink" >distance </span>between the <span class="p-lightBlue" >center</span> and <span class="p-lightPurple" >any point</span> 'inside' the <span class="p-lightYellow" >unit square</span>?
                                       
                                    </p>
                                </div>
                            </div>
                            <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center rotatingSectionGraph-1-1" id="rotatingSectionGraph-1-1-1" style="position: absolute; left: 200vw;">
                                <div class="our-text-block text-center" style="height: 400px; width: 100%; position: relative;">
                                    <div id="distance-between-center-and-any-point-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                                </div>
                            </div>
                            <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                                <div class="our-text-block text-center">
                                    <p>
                                        Take a guess. :D
                                    </p>
                                    <div class="mt-1 pb-2">
                                        <input type="range" min="0" max="1" value="0.2" step="0.001"  class="commonSlider mb-3" id="distance-between-center-and-point-inside-Slider">
                                        <div class="d-block d-sm-none"></div>
                                        <label id="distance-between-center-and-point-inside-SliderLabel" class="p-lightPink pl-sm-3 pl-0" for="distance-between-center-and-point-inside-Slider" style="min-width: 100px;" >0.8</label>
                                        <div>
                                            <p id="distance-between-center-and-point-inside-SliderInfo" class="text-small-ignus" style="color: hsl(0deg 0% 100%);">
                                                Point<span class="p-lightPurple">B</span> above is draggable. :)
                                            </p>
                                            
                                            <div id="gph3-checkAnswerButton" class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus" style="border: 1px solid hsla(var(--lightPinkH), var(--lightPinkS), var(--lightPinkL), 0.2); color: hsla(var(--lightPinkH), var(--lightPinkS), var(--lightPinkL), 1);">Check</div>
                                            <div id="reveal-sub-section-1-continueButton" class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus">Continue</div>
            
                                        </div>
                                    </div>
                                    
            
                                </div>
                            </div>
                            <div class="w-100 pl-4 pr-4 pb-5 pl-sm-5 pr-sm-5 d-flex justify-content-center reveal-sub-section-1 revealable">
                                <div class="mt-2 text-center">
                                    <p>
                                        It is about <span class="p-lightPink">0.38259785</span> ... To be exact, it is.
                                    </p>
                                    
                                    <div class="p-lightPink" style="font-size: 25px;">
                                        \(\color{#e3a0cd}{\frac{\color{#ffffff}{P}}{6}}\)
                                    </div>
                                
                                
                                    <br />
                                    <p>
                                        Yep. There's a <span class="p-White">P</span> here too. <br />
                                    </p>
                                </div>
                                

                            </div>
                            
                        </div>
                        <div id="rotatingSection-1-2" class="rotatingSection-1" style="display: none">
                            <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center ">
                                <div class="our-text-block text-center pb-1">
                                    <p>
                                        Here is another problem. Try to guess this one.<br/>
                                        What do you think is the average <span class="p-lightPink" >distance </span>between a <span class="p-lightBlue" >corner</span> and <span class="p-lightPurple" >a point</span> on the <span class="p-lightYellow" >unit square</span>?
                                       
                                    </p>
                                
                                </div>
                            </div>
                            <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center rotatingSectionGraph-1-2" id="rotatingSectionGraph-1-2-1" style="position: absolute; left: 200vw;">
                                <div class="our-text-block text-center" style="height: 400px; width: 100%; position: relative;">
                                    <div id="distance-between-corner-and-point-on-square-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                                </div>
                            </div>
                            <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                                <div class="our-text-block text-center">
                                    <p>
                                        Take a guess. :D
                                    </p>
                                    <div class="mt-1 pb-2">
                                        <input type="range" min="0" max="2" value="0.2" step="0.001"  class="commonSlider mb-3" id="distance-between-corner-and-point-Slider">
                                        <div class="d-block d-sm-none"></div>
                                        <label id="distance-between-corner-and-point-SliderLabel" class="p-lightPink pl-sm-3 pl-0" for="distance-between-corner-and-point-Slider" style="min-width: 100px;" >0.8</label>
                                        <div>
                                            <p id="distance-between-corner-and-point-SliderInfo" class="text-small-ignus" style="color: hsl(0deg 0% 100%);">
                                                Point<span class="p-lightPurple">B</span> above is draggable. :)
                                            </p>
                                            
                                            <div id="gph4-checkAnswerButton" class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus" style="border: 1px solid hsla(var(--lightPinkH), var(--lightPinkS), var(--lightPinkL), 0.2); color: hsla(var(--lightPinkH), var(--lightPinkS), var(--lightPinkL), 1);">Check</div>
                                            <div id="reveal-sub-section-2-continueButton" class="commonButton p-2 pl-4 pr-4 mt-1 btn-ignus">Continue</div>
                                        </div>
                                    </div>
                                    
            
                                </div>
                            </div>
                            <div class="w-100 pl-4 pr-4 pb-5 pl-sm-5 pr-sm-5 d-flex justify-content-center reveal-sub-section-2 revealable">
                                <div class="mt-2 text-center">
                                    <p>
                                        It is about <span class="p-lightPink">0.76519571</span>...  To be exact, it is.
                                    </p>
                                    
                                    <div class="p-lightPink" style="font-size: 25px;">
                                        \(\color{#e3a0cd}{\frac{\color{#ffffff}{P}}{3}}\)
                                    </div>
                                
                                    <br />
                                    <p>
                                        There's a <span class="p-White">P</span> here too. <br />
                                    </p>
                                </div>
                                

                            </div>
                        </div>
                        
                    </div>
    
                    
    
                    <div  class="w-100 pl-4 pt-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center hidden-element reveal-1 revealable">
                        <div class="our-text-block text-center">
                            <p>
                                Let's branch out a little and delve into a problem unrelated to the unit square. A problem related to <span class="p-lightPurple" >Shrinking Random Walk</span> appeared in Tom Yuster's April 2017 Math Horizons piece (pp. 32-33)
                            </p>
                        </div>
                    </div>

                    <div  class="w-100 pl-4 pt-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center hidden-element reveal-1 revealable">
                        <div class="our-text-block text-center">
                            <p>
                                Consider a drunkard walking in a plane that starts at the <span class="p-White" >origin</span> and moves only in the east and north directions. He randomly chooses a direction and takes a step. The walk is such that the length of the <span class="p-lightOrange" >first step must be ‚àö2</span>, the length of the <span class="p-brightYellow" >second step must be ‚àö2/2</span>, the length of the <span class="p-lightGreen" >third step must be ‚àö2/4</span>, and <span class="p-lightBlue" >so on</span>.
                            </p>
                        </div>
                    </div>

                    <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center  hidden-element reveal-1 revealable">
                        <div class="our-text-block text-center" style="height: 500px; width: 100%; position: relative;">
                            <div id="random-shrinking-walk-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                            
                        </div>
                    </div>

                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-starting-block">
                            <p>Let us start with a innocent looking problem. Picture a <span class="p-lightYellow">unit square</span>, a perfect little 1x1 entity. Now, pick a point <span class="p-lightPurple">Q</span> at random along the square's boundary. What do you think is the <span class="p-lightPink">average distance</span> between this chosen <span class="p-lightPurple">point Q</span> and the very heart of our square, the <span class="p-lightBlue">center C</span>? </p>
                            <p> The average distance is the typical or middle value obtained when you add up all the distances between any two points then divide by the number of distances you added together. It's a way of finding a 'common' or 'usual' distance in a group of different distances.</p>
                        </div>
                        
                    </div>
    
                    <div class="w-100 pl-0 pr-0 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" style="height: 400px; width: 100%; position: relative;">
                            <div id="upc-measurement-graphHolder" style="display: inline-block; position: relative; width: 100%; height: 100%;"></div>
                        </div>
                    </div>
    
                    <div class="w-100 pl-4 pr-4 pl-sm-5 pr-sm-5 d-flex justify-content-center">
                        <div class="our-text-block text-center" id="gph1-explaining-block">
                            <p>Surprisingly, the answer is approximately <span class="p-lightPink" id="parameter-measured-parabolic-constant-by-4">0.5738967</span>... a peculiar figure, isn't it? This number equals <span class="p-White" >P</span>/4 where <span class="p-White" >P</span> is known as <span class="p-White" >Universal Parabolic Constant</span>. The value is around <span class="p-White" >2.295587149...</span> Hold on a second - a Parabolic Constant? But our question was about squares, wasn't it? <span class="p-White" >Where does the parabola come in? </span></p>
                        </div>
                    </div>



                    
                    <div class="w-100 pt-5 pb-5"></div>
                    <div class="w-100 pt-5 pb-5"></div>
                </main>

                <div style="position: fixed; height: 100%; width: 100vw; pointer-events: none; bottom: 0px; left: 0px; z-index: 5;" id="lottie-popper"></div>


            </div>

            
			
		</div>

        <!-- <audio id="backgroundMusic" src="assets/audio/grant-piano.mp3"></audio> -->
        <audio id="warmerMusic" src="assets/audio/state-change_confirm-up.ogg"></audio>
        <audio id="colderMusic" src="assets/audio/state-change_confirm-down.ogg"></audio>

		<script type="text/javascript">		
			darkmode = true

			if (darkmode) {
				document.body.style.background = 'black'
			}
		</script>

        <!-- <script type="text/javascript" src="https://cdn.jsdelivr.net/gh/prajwalsouza/viewX@latest/viewx.js"></script>  -->
        <script src="https://cdnjs.cloudflare.com/ajax/libs/bodymovin/5.7.6/lottie.min.js"></script>

        <script type="text/javascript" src="JS/viewx.js"></script> 

        <script>
            ignusTheme.baseHue = 201;
            ignusTheme.baseLightness = 70;
            ignusTheme.setUp();

            MathJax = {
                loader: {load: ['[tex]/color']},
                options: {
                    enableMenu: false
                }
            };
        </script>

        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/input/tex/extensions/color.js" charset="UTF-8"></script>

        <script src="/JS/universal-parabolic-constant/denovo-peptide-sequencing.js"></script>

        <!-- Firebase SDK -->
        <!-- <script src="https://www.gstatic.com/firebasejs/8.7.1/firebase-app.js"></script>
        <script src="https://www.gstatic.com/firebasejs/8.7.1/firebase-database.js"></script> -->

        <script>

            // const firebaseConfig = {
            //     apiKey: "AIzaSyB5RWSdCAumPvg78h53pjdI1Xvg5cx5Urk",
            //     authDomain: "parabolic-constant.firebaseapp.com",
            //     projectId: "parabolic-constant",
            //     storageBucket: "parabolic-constant.appspot.com",
            //     messagingSenderId: "1095465366835",
            //     appId: "1:1095465366835:web:2a4ad7759ad5948aab7a46"
            // };
          
            // // Initialize Firebase
            // const app = firebase.initializeApp(firebaseConfig);

            // var database = firebase.database();

            function uniqueIDGen() {
                sevenZBit = 78364164096;
                randomString = sevenZBit + parseInt(Math.random() * sevenZBit * 35)
                dateStringGen = new Date().getTime().toString(36)
                return (dateStringGen + "-" + randomString.toString(36))
            }

            denovoApp.sessionID = uniqueIDGen()
            var currentDate = new Date();



            function getUserDetails() {
                // Retrieve user agent
                var userAgent = navigator.userAgent;

                // Retrieve screen resolution
                var screenWidth = window.screen.width;
                var screenHeight = window.screen.height;

                // Retrieve browser dimensions
                var browserWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
                var browserHeight = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;

                // Retrieve user's preferred language
                var preferredLanguage = navigator.language;

                // Return an object with all the user details
                return {
                    userAgent: userAgent,
                    screenWidth: screenWidth,
                    screenHeight: screenHeight,
                    browserWidth: browserWidth,
                    browserHeight: browserHeight,
                    preferredLanguage: preferredLanguage
                };
            }

            dataToBeSaved = {}


            // function saveData() {
            //     if (window.location.href != "http://127.0.0.1:5500/universal-parabolic-constant.html") {
            //         dataToBeSaved = {}

            //         dataToBeSaved.interactions = denovoApp.interactions
            //         dataToBeSaved.dateTimeString = currentDate.toString()
            //         dataToBeSaved.id = denovoApp.sessionID
            //         dataToBeSaved.graph6Trials = gph6.trialsAdded
            //         var userDetails = getUserDetails();
            //         dataToBeSaved.userDetails = userDetails;

            //         database.ref('sessions/' + denovoApp.sessionID).set(dataToBeSaved)
            //             .then(function() {
            //                 // console.log("Data saved successfully."); 
            //             })
            //             .catch(function(error) {
            //                 // console.log("Error writing data:", error);
            //             });
            //     }
            // }


            denovoApp.interactionNumber += 1
            dateTimeNow = new Date()
            denovoApp.interactions[denovoApp.interactionNumber] = {
                dateTime : dateTimeNow.toString(),
                "type": "page load"
            }

            // saveData()


          


            


        </script>

        <script>
            window.addEventListener('scroll', function(e) {
                if (window.scrollY < 900) {
                    document.getElementById('upc-header').style.display = "none";
                    document.body.style.background = "black"
                }
                else if (window.scrollY >= 900 && window.scrollY < 1500) {
                    document.getElementById('upc-header').style.display = "block";
                    document.getElementById('upc-header').style.opacity = viewX.linearValue(900, 1500, 0, 1, window.scrollY)
                    document.body.style.background = "var(--generalBackground)"
                }
                else {
                    document.getElementById('upc-header').style.display = "block";
                    document.getElementById('upc-header').style.opacity = 1
                    document.body.style.background = "var(--generalBackground)"
                }

                // denovoApp.renderMenus()
  
            });

            document.getElementById('upc-header').style.display = "none";
            
            document.getElementById('upc-menus').style.display = "none"

            
            denovoApp.menuHighlightDetection = function(targetDiv) {
                const observer = new IntersectionObserver((entries, observer) => {
                    entries.forEach(entry => {
                        if (entry.isIntersecting) {
                            document.getElementById('upc-menus').style.display = "none"
                            allMenuState = document.getElementById('upc-menus').style.display
                            if (allMenuState == "none") {
                                document.getElementById('currentMenuName').innerText = nameMenuMaps[entry.target.id].name
                            }
                            
                            nameMenuMaps[entry.target.id].revealed = true;
                            denovoApp.currentMenuSelected = nameMenuMaps[entry.target.id].name
                            denovoApp.renderMenus()

                            
                        }
                    });
                });

                observer.observe(targetDiv);
            }

            nameMenuMaps = {
                'upc-measurement-graphHolder': {name: 'What is the problem?', revealed: false},
                'distance-between-any-two-points-graphHolder': {name: 'Distance between any two points', revealed: false}

            }

            for (divIndex  = 0; divIndex < Object.keys(nameMenuMaps).length; divIndex++) {
                divName = Object.keys(nameMenuMaps)[divIndex]
                menuName = nameMenuMaps[divName]

                denovoApp.menuHighlightDetection(document.getElementById(divName))
            }

            denovoApp.scrollToDiv = function(divName) {
                document.getElementById(divName).scrollIntoView({behavior: "smooth", block: "start", inline: "nearest"});

                denovoApp.currentMenuSelected = nameMenuMaps[divName].name
                denovoApp.headerClick();
            }

            denovoApp.headerClick = function() {
                allMenuState = document.getElementById('upc-menus').style.display
                if (allMenuState != "none") {
                    document.getElementById('upc-menus').style.display = "none"
                    document.getElementById('currentMenuName').innerText = denovoApp.currentMenuSelected
                    // console.log(denovoApp.currentMenuSelected)
                }
                else {
                    document.getElementById('upc-menus').style.display = "block"
                    document.getElementById('currentMenuName').innerText = ""
                }

                denovoApp.renderMenus()
                
            }

            denovoApp.renderMenus = function() {
                menuInnerHTML = ""
                menusAdded = {}

                // document.getElementById('currentMenuName').innerText = ""
                for (divIndex  = 0; divIndex < Object.keys(nameMenuMaps).length; divIndex++) {
                    divName = Object.keys(nameMenuMaps)[divIndex]
                    menuOptions = nameMenuMaps[divName]
                    // console.log(typeof(menusAdded[menuOptions]), menuOptions)
                    if (menuOptions.revealed && typeof(menusAdded[menuOptions.name]) == 'undefined') {
                        // console.log(denovoApp.currentMenuSelected, menuOptions.name)
                        menuColor = denovoApp.currentMenuSelected == menuOptions.name ? "darkgrey" : "grey";
                        tagType = denovoApp.currentMenuSelected == menuOptions.name ? "b" : "span";
                        // console.log(menuColor)
                        menuInnerHTML += "<" + tagType + " class='pl-0 p-0 text-small-ignus' onclick='denovoApp.scrollToDiv(\"" + divName + "\")' style='color: " + menuColor + "; cursor: pointer;'>" + menuOptions.name + "</" + tagType + "><br/>"
                        menusAdded[menuOptions.name] = divName
                    }
                }

                document.getElementById('upc-menus').innerHTML = menuInnerHTML

                
            }


        </script>



	</body>
<html>

